lumicks/pylake/kymo.py

```diff
@@ -11,6 +12,14 @@
 from .detail.timeindex import to_timestamp
 
 
+def _default_line_time_factory(self):
+    """Line time in seconds"""
+    if self.timestamps.shape[1] > 1:
+        return (self.timestamps[0, 1] - self.timestamps[0, 0]) / 1e9
+    else:
+        raise RuntimeError("Line time is not defined for kymograph with only a single line")
```

Comment:
I would have no idea what to do with that error message. Perhaps link to a definition of `line time`, or explain briefly, so that the uninitiated hacker at least has a fighting chance to understand why this logic is correct?

From reading the Kymo tutorial, I get the idea that it measures the timedelta between two lines, but then I would expect the delta for one line to just be 0. I must be missing something fundamental here why that's that not an acceptable return value.

Reply:
I think alerting the user that it's one line is better than quietly returning zero. When recording the `Kymo` they still set an intended line time. I would say that this would still be the intended return value.

However, we have no way of knowing whether what we got is one completed line or one incomplete line (the latter is more likely even). Hence there is no way of knowing the actual linetime that was intended.

I do agree that the message could use a touch up though.
```python
        raise RuntimeError("This kymograph consists of only a single line. It is not possible to "
                           "determine the kymograph line time for a kymograph consisting only of a "
                           "single line.")
```